plugins/sentinelone/komand_sentinelone/util/api.py

```diff
@@ -68,19 +100,10 @@ def search_agents(
 
                 # Normalize casing if specified
                 if not case_sensitive:
-                    if search == "computerName":
-                        agents = [agent_details.lower(), agent_details.upper()]
-                    if search == "uuid":
-                        agents = [agent_details.lower()]
+                    agents = set_agents_array(search, agent_details, agents)
 
                 for agent in agents:
-                    endpoint = f"{self.url}web/api/v{api_version}/agents?{search}={agent}"
-                    output = requests.get(endpoint, headers=self.token_header)
-
-                    if output.status_code == 200 and output.json().get("pagination", {}).get("totalItems", 0) >= 1:
-                        agents_data = output.json().get("data", [])
-                        if agents_data:
-                            results.append(agents_data[0])
+                    get_agents_data(self, agent, api_version, search, results)
```

Comment:
Don't think we should mutate the existing results array. We can return the new one as a result of that methods.

Why you are passing the `self` here? Move that methods to the `SentineloneAPI` class and it will be allowed to get the API class instance.

```suggestion
results = self.get_agents_data(agent, api_version, search)
```